Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs

```diff
@@ -37,15 +42,51 @@ public class WindowsMixedRealityDeviceManager : BaseInputDeviceManager
             uint priority = DefaultPriority,
             BaseMixedRealityProfile profile = null) : base(registrar, inputSystem, name, priority, profile) { }
 
+        /// <inheritdoc/ >
+        public override bool CheckCapability(MixedRealityInputCapabilities capability)
+        {
+            if (WindowsApiChecker.UniversalApiContractV8_IsAvailable)
+            {
+                // Windows 10 1903 contains an API that allows to check input type support.
+#if WINDOWS_UWP
+                switch (capability)
+                {
+                    case MixedRealityInputCapabilities.ArticulatedHand:
+                    case MixedRealityInputCapabilities.GGVHand:
+                        return WindowsInputSpatial.SpatialInteractionManager.IsSourceKindSupported(WindowsInputSpatial.SpatialInteractionSourceKind.Hand);
+
+                    case MixedRealityInputCapabilities.MotionController:
+                        return WindowsInputSpatial.SpatialInteractionManager.IsSourceKindSupported(WindowsInputSpatial.SpatialInteractionSourceKind.Controller);
+                }
+#endif // WINDOWS_UWP
+            }
+            else
+            {
+                // Pre-Windows 10 1903.
+                if (!UnityEngine.XR.WSA.HolographicSettings.IsDisplayOpaque)
+                {
+                    // HoloLens supports GGV hands
+                    return (capability == MixedRealityInputCapabilities.GGVHand);
+                }
+                else
+                {
+                    // Windows Mixed Reality Immersive devices support motion controllers
+                    return (capability == MixedRealityInputCapabilities.MotionController);
+                }
+            }
+
+                return false;
+        }
+
 #if UNITY_WSA
 
-        /// <summary>
-        /// The initial size of interactionmanagerStates.
-        /// </summary>
-        /// <remarks>
-        /// This value is arbitrary but chosen to be a number larger than the typical expected number (to avoid
-        /// having to do further allocations).
-        /// </remarks>
+            /// <summary>
```

Comment:
I think extra tabs were added here by mistake (this should have the same indent as MaxInteractionSourceStates)